### PR TITLE
Add the action "Group" in the assignment rule engine habilitations to user

### PR DESCRIPTION
**Pull request by amma35**
Add the action "Group" in the assignment rule engine habilitations to a user

see #1138

### CODE REVIEW

File: inc/ruleright.class.php [hunk 1]
```diff
@@ -151,6 +152,9 @@ function executeActions($output, $params) {
                         $continue                   = false;
                         $output_src["_stop_import"] = true;
                         break;
+					case "groups_id" :
+						$output["groups_id"] = $action->fields["value"];
```
Comment (remicollet) [inline inc/ruleright.class.php]: virtual fields (not part of the glpi_users table should be _ prefixed.

This also mean a single group can be add, which sounds not correct

File: inc/user.class.php [hunk 2]
```diff
@@ -599,7 +599,7 @@ function post_addItem() {
 
       $this->syncLdapGroups();
       $this->syncDynamicEmails();
-      $rulesplayed = $this->applyRightRules();
+      $rulesplayed = $this->applyRightRules($this->getID());
```
Comment (remicollet) [inline inc/user.class.php]: Why is this needed, $this-getID is available in the applyRightRules method

File: inc/user.class.php [hunk 3]
```diff
@@ -984,6 +984,13 @@ function applyRightRules() {
                   $right->add($retr_profile);
                }
             }
+			
+			if (isset($this->input["groups_id"])) {
+               $group_user = new Group_user();
+               $input['users_id'] = $users_id;
+               $input['groups_id'] = $this->input["groups_id"];
+               $group_user->add($input);
```
Comment (remicollet) [inline inc/user.class.php]: The new relation have to be set as dynamic, to be clean when rules change.

But this raise the issue as dynamic group are clean during ldap sync...

**Comment by remicollet** [comment]
Dynamic group (in $this->input['_groups']) are synced in syncLdapGroups method.

This should be used (and probably adapted, as for now, this is only executed for LDAP users).

BTW, I will be curious to see an example of rules you want to write which cannot be managed by already implemented group sync.

File: inc/ruleright.class.php (hunk 1 above)
Comment (amma35) [inline inc/ruleright.class.php]: Ok for _prefixed

**Comment by orthagh** [comment]
Hello @amma35 

Do you plan to update this pr (CS mainly) or i should close it ?

**Comment by orthagh** [comment]
I archive this PR (no answer)
